docs/tutorials/2. Real-time analysis.md

```diff
@@ -0,0 +1,225 @@
+
+1. Getting a live audio stream from the user's microphone is done as above (using `navigator.mediaDevices.getUserMedia`), as is setting up the audio graph, connecting the nodes.
+2. When setting up the audio graph, we have to create an [`AudioWorkletNode`](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorkletNode) instead of a `ScriptProcessorNode`. This will allow us to run our custom audio code on its own rendering thread, separate from the UI. The following code returns one such node:
+
+```javascript
+const workletProcessorCode = ["https://cdn.jsdelivr.net/npm/essentia.js@0.1.0/dist/essentia-wasm.module.js", "https://cdn.jsdelivr.net/npm/essentia.js@0.1.0/dist/essentia.js-core.es.js", "essentia-worklet-processor.js"];
+
+export async function createEssentiaNode (audioCtx) {
+  try {
+    let concatenatedCode = await URLFromFiles(workletProcessorCode)
+    await audioCtx.audioWorklet.addModule(concatenatedCode); // add our custom code to the worklet scope
+  } catch(e) {
+    console.log(e);
+  }
+  return new AudioWorkletNode(audioCtx, 'essentia-worklet-processor');
+}
+```
+
+3. The async `URLFromFiles()` function ([by Paul Adenot](https://github.com/padenot/ringbuf.js/blob/master/example/utils.js)) is used to fetch and concatenate our custom code together with the `essentia.js` library. This is necessary because the `fetch` or `XHR` APIs are not available in the Worklets' scope, and ES6 module imports are only available in Chrome. This code allows our custom processing to work also on Firefox and Edge.
+
+```javascript
+function URLFromFiles(files) {
+    const promises = files
+        .map((file) => fetch(file)
+        .then((response) => response.text()));
+
+    return Promise
+        .all(promises)
+        .then((texts) => {
+            const text = texts.join('');
+            const blob = new Blob([text], {type: "application/javascript"});
+
+            return URL.createObjectURL(blob);
+        });
+}
+```
+
+4. This is used in the audio graph setup function as follows:
+
+```javascript
+    micNode = audioContext.createMediaStreamSource(gumStream);
+    // ...
+    // create essentia node only once (avoid registering processor repeatedly)
+    if (!essentiaNode) {
+        essentiaNode = await createEssentiaNode(audioContext);
+    }
+
+    micNode.connect(essentiaNode);
+```
+
+5. Finally, this is our custom audio code, defined inside _essentia-worklet-processor.js_ (which we used in `createEssentiaNode`). It inherits from [`AudioWorkletProcessor`](https://developer.mozilla.org/en-US/docs/Web/API/AudioWorkletProcessor):
+
+```javascript
+let essentia = new Essentia(Module); // where Module is EssentiaWASM object we concatenated to this code by using URLFromFiles
```

Comment:
This is a hacky solution from our end. It would be nice to have a generic one without this manual hack. How about having a generic ES6 import example here? So, it is not confusing to the users. We can mention this cross-browser solution at the end by hyperlinking to the example in our repo or even add a section in the developer notes.